net/mwan3/files/usr/sbin/mwan3track

```diff
@@ -345,6 +365,7 @@ main() {
 				echo "online" > $MWAN3TRACK_STATUS_DIR/$INTERFACE/STATUS
 				score=$((down+up))
 			elif [ $score -le $up ]; then
+				connecting
```

Comment:
It looks like this will cause it to trigger a "connecting" event with each successful ping when it is in the down state, no?

Reply:
The connecting function is called on every check if we are in tear up (get connecting).
That`s right.
But if the connection is setting up, then the hotplug event is only called once.
The function [set the STATUS to connecting](https://github.com/openwrt/packages/pull/14996/files#diff-df4fb4fa188ff19ed301dde4878253e6b8931bed61bcc92c7c5de32beec9840eR143) and so does not do another hotplug event because we [check STATUS](https://github.com/openwrt/packages/pull/14996/files#diff-df4fb4fa188ff19ed301dde4878253e6b8931bed61bcc92c7c5de32beec9840eR141) if we have already entered this code section.

So I think this does not happen.

Reply:
I see. `mwan3track` should really be refactored to use a state machine architecture, to make this sort of thing more clear,  but that is a larger project. This P/R looks fine. Approved.

Reply:
I agree with you completely.
The question is also whether we should do it in native shell again or perhaps use lua.
I've been toying with the idea for a while.

Reply:
Native shell is nice in that is accessible to more developers, but it would be great to do it in a language that allows multithreading. Having separate mwan3track processes for each device is a large part of mwan3's memory footprint.

If in Lua, we could hook in directly to netifd ubus notifications and get rid of the separate hotplug script too.